components/script/dom/xmlhttprequest.rs

```diff
@@ -901,6 +901,7 @@ impl XMLHttpRequest {
 
                 // Subsubsteps 5-7
                 self.send_flag.set(false);
+                *self.response_url.borrow_mut() = self.request_url.borrow().as_ref().unwrap().serialize_no_fragment();
```

Comment:
This value needs to come from the `final_url` field of `Metadata` instead, since that is the post-redirection URL.